src/lizaalert/courses/models.py

```diff
@@ -97,27 +122,54 @@ class Meta:
     def __str__(self):
         return f"Course {self.title}"
 
-    def finish(self, user):
-        """Закончить весь курс."""
-        CourseProgressStatus.objects.update_or_create(
-            user=user, course=self, usercourseprogress=CourseProgressStatus.ProgressStatus.FINISHED
-        )
-
     def current_lesson(self, user):
-        """Вернуть queryset текущего урока."""
+        """Return queryset of the current lesson."""
         finished_lessons = LessonProgressStatus.objects.filter(
-            user=user, userlessonprogress=LessonProgressStatus.ProgressStatus.FINISHED
+            user=user, progress=LessonProgressStatus.ProgressStatus.FINISHED
         ).values_list("lesson_id", flat=True)
 
-        return (
-            Lesson.objects.filter(chapter__course=self, status=Lesson.LessonStatus.PUBLISHED)
-            .exclude(id__in=finished_lessons)
-            .annotate(ordering=F("chapter__order_number") + F("order_number"))
-            .order_by("ordering")
+        lesson_queryset = Lesson.objects.filter(chapter__course=self, status=Lesson.LessonStatus.PUBLISHED).annotate(
+            ordering=F("chapter__order_number") + F("order_number")
+        )
+        current_lesson_queryset = lesson_queryset.exclude(id__in=finished_lessons).order_by("ordering")[:1]
+
+        if not current_lesson_queryset.exists():
+            return lesson_queryset.order_by("-ordering")[:1]
+
+        return current_lesson_queryset
+
+    @property
+    def is_available(self):
+        """Проверить доступность курса."""
+        return timezone.now().date() >= self.start_date
+
+    def subscribe(self, user):
+        """Подписать пользователя на данный курс."""
+        subscription, created = Subscription.objects.get_or_create(user=user, course=self)
+        if not created:
+            raise AlreadyExistsException({"detail": "Already enrolled."})
+        return subscription
+
+    def finish(self, user):
+        super().finish(user)
+        progress_status, created = Subscription.objects.get_or_create(
+            user=user, course=self, defaults={"status": Subscription.Status.COMPLETED}
+        )
+        if not created:
+            progress_status.status = Subscription.Status.COMPLETED
+            progress_status.save()
+
+    def activate(self, user):
+        super().activate(user)
+        progress_status, created = Subscription.objects.get_or_create(
+            user=user, course=self, defaults={"status": Subscription.Status.IN_PROGRESS}
         )
+        if not created:
+            progress_status.status = Subscription.Status.IN_PROGRESS
+            progress_status.save()
 
 
-class Chapter(TimeStampedModel, order_number_mixin(CHAPTER_STEP, "course")):
+class Chapter(TimeStampedModel, order_number_mixin(CHAPTER_STEP, "course"), status_update_mixin(parent="course")):
```

Comment:
лучше добавить магическую запятую в список, чтобы каждый класс был на отдельной 

```suggestion
class Chapter(
    TimeStampedModel,
    order_number_mixin(CHAPTER_STEP, "course"),
    status_update_mixin(parent="course"),
):
```